Implemented linear and radial gradients for fill and stroke colors.

winsock opened the pull request:
I have not tested this extensively. It was a quick hack of the code so I didn't have to manually import all of the SVGs with gradients into Android Studio. For the 41 SVGs I tested it with it worked properly.
One outstanding issue is that I didn't bother with making sure indentation worked properly.

winsock committed:
Implemented linear and radial gradients for fill and stroke colors.

Fixed radial gradients.

Enable transforms for gradients.

Radial transforms not fully supported. Not sure how to deal with scale for them.

Parse the transforms as floats

Update main.js

Fix item tag generation and substituteUseRef

This should be a return

Parse the style attribute on the stop tag.

Adobe likes making stop-color and stop-opacity attributes. Looks like Inkscape use stye tags.

Typo with an extra parentheses

Fix the extra tag close

Bump version number

So it doesn't load an old version from the browser cache.

yuraj11 commented:
It looks OK but It does not work correctly on this - could you please check It?
Also when I try to build the project It shows me error:
> Error while processing C:\Users\Yuraj\Repos\Experiments2\app\src\main\res\drawable\aa.xml : Start point cannot equalendpoint

[problematic_drawing.zip](https://github.com/inloop/svg2android/files/1445774/drawing.zip)

yuraj11 commented:
@winsock Could you please check my SVG and try convert with your changes? Because It does not correctly generates `<gradient` block

ddobrev commented:
@winsock thank you for your work. Could you please finish it?
@yuraj11 in case @winsock is unable to continue, could you please finish his work? This feature is really important.

winsock commented:
@yuraj11 @ddobrev I apologize for the really late response for addressing these issues. I have been slammed at work with a project leaving almost no free time.

Your test SVG now renders properly. Apparently there were a couple of issues, substituteUseRef wasn't copying children and Inkscape was using style attributes for stop-color and opacity unlike Adobe Illustrator setting those as their own attributes.

winsock commented:
Just rebased and squashed as well to make this merge cleaner
